src/libstd/sys/windows/printing/msvc.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use sys_common::backtrace::*;
+use ffi::CStr;
+use str;
+use dynamic_lib::DynamicLibrary;
+use super::SymFromAddrFn;
+use io::prelude::*;
+
+pub fn print(w: &mut Write, i: isize, addr: u64, lib: &DynamicLibrary) -> io::Result<()> {
+    let SymFromAddr = sym!("SymFromAddr", SymFromAddrFn);
+
+    try!(write!(w, "  {:2}: {:#2$x}", i, addr, HEX_WIDTH));
+    let mut info: SYMBOL_INFO = unsafe { intrinsics::init() };
+    info.MaxNameLen = MAX_SYM_NAME as libc::c_ulong;
+    // the struct size in C.  the value is different to
+    // `size_of::<SYMBOL_INFO>() - MAX_SYM_NAME + 1` (== 81)
+    // due to struct alignment.
+    info.SizeOfStruct = 88;
+
+    let mut displacement = 0u64;
+    let ret = SymFromAddr(process, addr as u64, &mut displacement,
+                          &mut info);
+
+    if ret == libc::TRUE {
+        try!(write!(w, " - "));
+        let ptr = info.Name.as_ptr() as *const libc::c_char;
+        let bytes = unsafe { CStr::from_ptr(ptr).to_bytes() };
+        match str::from_utf8(bytes) {
+            Ok(s) => try!(demangle(w, s)),
+            Err(..) => try!(w.write_all(&bytes[..bytes.len()-1])),
+        }
+    }
+    try!(w.write_all(&['\n' as u8]));
```

Comment:
Hm I'm not entirely sure why this wasn't refactored yet, but can you also refactor this to call the common `output` routine shared by Unix as well?